File: Same Binary Tree - Leetcode 100/Same Binary Tree - Leetcode 100.java
public class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        return isSame(p, q);
    }

    private boolean isSame(TreeNode p, TreeNode q) {
        if (p == null && q == null) return true;
        if (p == null || q == null) return false;
        if (p.val != q.val) return false;

        return isSame(p.left, q.left) && isSame(p.right, q.right);
    }
}

File: Same Binary Tree - Leetcode 100/Same Binary Tree - Leetcode 100.js
var isSameTree = function(p, q) {
    return isSame(p, q);
};

function isSame(p, q) {
    if (!p && !q) return true;
    if (!p || !q) return false;
    if (p.val !== q.val) return false;

    return isSame(p.left, q.left) && isSame(p.right, q.right);
}
